linker/shared/src/main/scala/org/scalajs/linker/frontend/BaseLinker.scala

```diff
@@ -189,9 +190,19 @@ private[frontend] object BaseLinker {
 
     val allMethods = methods ++ syntheticMethodDefs
 
-    val kind =
-      if (classInfo.isModuleAccessed) classDef.kind
-      else classDef.kind.withoutModuleAccessor
+    val kind = {
+      import ClassKind._
+      classDef.kind match {
+        case kind if kind.isJSType && !classInfo.isInstantiated => AbstractJSType
```

Comment:
Shouldn't this be `isAnySubclassInstantiated`?

Reply:
For JS types, they're basically equivalent. But it doesn't matter anyway, the whole approach of patching up the `kind` is wrong, as painfully shown by the CI. :sweat_smile:

Reply:
Seems like we need the equivalent of this:
https://github.com/scala-js/scala-js/blob/2e8549c49716218ca1be18ea854c2fa66738d4f6/linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/SJSGen.scala#L235-L241

Here:
https://github.com/scala-js/scala-js/blob/2e8549c49716218ca1be18ea854c2fa66738d4f6/linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/ClassEmitter.scala#L957-L964